cdap-common/src/main/java/co/cask/cdap/common/namespace/AbstractNamespaceClient.java

```diff
@@ -37,14 +37,14 @@
   private static final Gson GSON = new Gson();
 
   @Override
-  public void delete(Id.Namespace namespaceId) throws Exception {
-    URL url = resolve(String.format("unrecoverable/namespaces/%s", namespaceId.getId()));
+  public void delete(NamespaceId namespaceId) throws Exception {
```

Comment:
Not sure. I'm all for breaking backwards compatibility in 4.0 for cdap-clients (to save on work on our end and also maintaining the old code), but I think we should have a bigger discussion before doing that.